application/models/services/SurveysGroupCreator.php

```diff
@@ -51,7 +55,24 @@ public function __construct(
      */
     public function save()
     {
-        $this->surveysGroup->attributes = $this->request->getPost('SurveysGroups');
+        $attributes = $this->request->getPost('SurveysGroups');
+
+        // Check if the parent group is valid
+        if (!empty($attributes['parent_id'])) {
+            $parentId = $attributes['parent_id'] ;
+            /* Check permission */
+            $availableParents = $this->surveysGroup->getParentGroupOptions();
```

Comment:
The getParentGroupOptions have more code than https://github.com/LimeSurvey/LimeSurvey/blob/76bb242b4224d0ae58f11a50d6c1f6adac58d4fd/application/models/SurveysGroups.php#L347

(we still need a scope for permission … )